log/src/main/java/br/com/codenation/log/service/implementations/LogService.java

```diff
@@ -1,36 +1,120 @@
-        return Collections.emptyList();
+    public List<LogDTO> listarComFiltros(Ambiente ambiente,
+                                         Nivel nivel,
+                                         String descricao,
+                                         String origem,
+                                         Ordenacao ordenacao) {
+
+        if (Objects.nonNull(nivel))
+            return filtraPorNivel(ambiente, nivel, ordenacao);
+
+        if (Objects.nonNull(descricao))
+            return filtraPorDescricao(ambiente, descricao, ordenacao);
+
+        if (Objects.nonNull(origem))
+            return filtraPorOrigem(ambiente, origem, ordenacao);
+
+        return filtraPorAmbiente(ambiente);
+    }
+
+    private List<LogDTO> filtraPorAmbiente(Ambiente ambiente) {
+        return mapLogProjectionToLogDTO(repository.findAllByAmbiente(ambiente.toString()));
+    }
+
+    private List<LogDTO> filtraPorNivel(Ambiente ambiente, Nivel nivel, Ordenacao ordenacao) {
+        List<LogDTO> logs = mapLogProjectionToLogDTO(repository.findAllByAmbienteAndNivel(ambiente.toString(), nivel.toString()));
+        return ordenaLista(logs, ordenacao);
+    }
+
+    private List<LogDTO> filtraPorDescricao(Ambiente ambiente, String descricao, Ordenacao ordenacao) {
+        List<LogDTO> logs = mapLogProjectionToLogDTO(repository.findAllByAmbienteAndDescricao(ambiente.toString(), descricao));
+        return ordenaLista(logs, ordenacao);
+    }
+
+    private List<LogDTO> filtraPorOrigem(Ambiente ambiente, String origem, Ordenacao ordenacao) {
+        List<LogDTO> logs = mapLogProjectionToLogDTO(repository.findAllByAmbienteAndOrigem(ambiente.toString(), origem));
+        return ordenaLista(logs, ordenacao);
+    }
+
+    private List<LogDTO> ordenaLista(List<LogDTO> logs, Ordenacao ordenacao) {
+        if (Objects.isNull(ordenacao))
+            return logs;
+
+        switch (ordenacao) {
+            case FREQUENCIA:
+                logs.sort(Comparator.comparingLong(LogDTO::getFrequencia));
+                break;
+            case NIVEL:
+                logs.sort(Comparator.comparing(log -> log.getNivel().name()));
+                break;
+        }
+
+        return logs;
+    }
+
+    private List<LogDTO> mapLogProjectionToLogDTO(List<LogProjection> projections) {
+        return projections.stream()
+                .map(this::mapLogProjectionToLogDTO)
+                .collect(Collectors.toList());
+    }
+
+    private LogDTO mapLogProjectionToLogDTO(LogProjection projection) {
+        Usuario usuario = usuarioRepository.findById(projection.getUsuarioId()).orElse(null);
+        PayloadDTO payloadDTO = stringToPayloadDTO(projection.getPayload());
+        Ambiente ambiente = Ambiente.valueOf(projection.getAmbiente());
+        Nivel nivel = Nivel.valueOf(projection.getNivel());
+
+        return new LogDTO(ambiente, nivel, payloadDTO, projection.getFrequencia(), usuario);
+    }
+
+    private PayloadDTO stringToPayloadDTO(String json) {
```

Comment:
```suggestion
    private Optional<PayloadDTO> stringToPayloadDTO(String json) {
```
Como pode retornar null nesse método é legal usar o Optional para evitar Null Pointer Exception.